src/hooks/useQuizTrackingAndCbEvents.ts

```diff
@@ -0,0 +1,183 @@
+/* eslint-disable max-params */
+/* eslint-disable @typescript-eslint/naming-convention */
+import ConstructorIOClient from '@constructor-io/constructorio-client-javascript';
+import { GetBrowseResultsResponseData } from '@constructor-io/constructorio-client-javascript/lib/types';
+import { useCallback, useEffect } from 'react';
+import { ActionAnswerQuestion, QuestionTypes } from '../components/CioQuiz/actions';
+import { QuizAPIReducerState } from '../components/CioQuiz/quizApiReducer';
+import { ResultsPageOptions } from '../components/Results/Results';
+import { isFunction } from '../utils';
+
+const useQuizTrackingAndCbEvents = (
+  cioClient: ConstructorIOClient,
+  quizApiState: QuizAPIReducerState,
+  resultsPageOptions: ResultsPageOptions,
+  dispatchLocalState: React.Dispatch<ActionAnswerQuestion>
+) => {
+  const { onAddToCartClick, onQuizResultClick, onQuizResultsLoaded } = resultsPageOptions;
+  // Quiz Next button click
+  const quizNextHandler = useCallback(
+    (payload?: any) => {
+      switch (quizApiState.quizCurrentQuestion?.next_question.type) {
+        case QuestionTypes.Cover:
+          dispatchLocalState({
+            type: QuestionTypes.Cover,
+            payload: {
+              isLastQuestion: quizApiState?.quizCurrentQuestion?.is_last_question,
+            },
+          });
+          break;
+        case QuestionTypes.OpenText:
+          dispatchLocalState({
+            type: QuestionTypes.OpenText,
+            payload: {
+              questionId: quizApiState?.quizCurrentQuestion.next_question.id,
+              input: payload,
+              isLastQuestion: quizApiState?.quizCurrentQuestion.is_last_question,
+            },
+          });
+          break;
+        case QuestionTypes.SingleSelect:
+        case QuestionTypes.MultipleSelect:
+          dispatchLocalState({
+            type:
+              quizApiState?.quizCurrentQuestion.next_question.type === QuestionTypes.SingleSelect
+                ? QuestionTypes.SingleSelect
+                : QuestionTypes.MultipleSelect,
+            payload: {
+              questionId: quizApiState?.quizCurrentQuestion.next_question.id,
+              input: payload,
+              isLastQuestion: quizApiState?.quizCurrentQuestion.is_last_question,
+            },
+          });
+          break;
+
+        default:
+          break;
+      }
+    },
+    [quizApiState, dispatchLocalState]
+  );
+
+  const quizBackHandler = useCallback(() => {
+    if (dispatchLocalState) {
+      dispatchLocalState({ type: QuestionTypes.Back });
+    }
+  }, [dispatchLocalState]);
+
+  // Quiz result add to cart tracking event
+  const addToCartClickHandler = useCallback(
```

Comment:
Why is this `useCallback` instead of a normal function? 🤔

Reply:
To avoid re initiating the function on each render or each time the hook gets called. This way the function only get reinitiated if any of it's dependency array items gets updated. like `quizApiState` for example. Happy to go over this in detail in our call